### PR TITLE
Change zipkin V1 conversion to use pdata, removes one extra step

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: pkg/translator/zipkin/zipkinv1/thrift.go [hunk 1]
```diff
@@ -20,16 +20,14 @@ import (
 	"encoding/binary"
 	"errors"
 	"fmt"
+	"go.opentelemetry.io/collector/pdata/pcommon"
```
Comment (dmitryax) [inline pkg/translator/zipkin/zipkinv1/thrift.go:23]: This import should be moved down

File: pkg/translator/zipkin/zipkinv1/status_code.go [hunk 2]
```diff
@@ -80,15 +82,18 @@ func (m *statusMapper) ocStatus() *tracepb.Status {
 		if s.codePtr != nil {
 			code = *s.codePtr
 		}
-		return &tracepb.Status{
-			Code:    code,
-			Message: s.message,
+		switch code {
+		case trace.StatusCodeOK:
+			dest.SetCode(ptrace.StatusCodeUnset)
```
Comment (dmitryax) [inline pkg/translator/zipkin/zipkinv1/status_code.go:87]: Shouldn't this be `ptrace.StatusCodeOk`?

File: pkg/translator/zipkin/zipkinv1/thrift.go (hunk 1 above)
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/thrift.go:23]: :) I did the PR from my personal laptop where I don't have the auto-format enabled, sorry.

File: pkg/translator/zipkin/zipkinv1/status_code.go (hunk 2 above)
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/status_code.go]: Fixed bunch of status issues.

File: pkg/translator/zipkin/zipkinv1/json.go [hunk 3]
```diff
@@ -106,215 +97,151 @@ type binaryAnnotation struct {
...
-	pbAttrib := &tracepb.AttributeValue{}
+func parseAnnotationValue(value string, parseStringTags bool) pcommon.Value {
+	val := pcommon.NewValueEmpty()
 
 	if parseStringTags {
 		switch zipkin.DetermineValueType(value) {
 		case pcommon.ValueTypeInt:
 			iValue, _ := strconv.ParseInt(value, 10, 64)
-			pbAttrib.Value = &tracepb.AttributeValue_IntValue{IntValue: iValue}
+			val.SetInt(iValue)
```
Comment (dmitryax) [inline pkg/translator/zipkin/zipkinv1/json.go:219]: Not a big deal but curious why not `return pcommon.NewValueInt(iValue)`?

**Review by dmitryax — COMMENTED** [review]
I'll do another round of review later today. For now only nits

File: pkg/translator/zipkin/zipkinv1/json.go [hunk 4]
```diff
@@ -106,215 +97,151 @@ type binaryAnnotation struct {
...
+			span:     ocSpan,
+			endpoint: endpoint,
 		})
 	}
 
-	return zipkinToOCProtoBatch(ocSpansAndParsedAnnotations)
+	return zipkinToTraces(ocSpansAndParsedAnnotations)
 }
 
 type ocSpanAndParsedAnnotations struct {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/json.go:123]: fixed :)

File: pkg/translator/zipkin/zipkinv1/json.go [hunk 5]
```diff
@@ -106,215 +97,151 @@ type binaryAnnotation struct {
...
-		}
-		ocSpan.Attributes.AttributeMap[tracetranslator.TagSpanKind] =
-			&tracepb.AttributeValue{Value: &tracepb.AttributeValue_StringValue{
-				StringValue: &tracepb.TruncatableString{Value: string(extendedKind)},
-			}}
-	}
-}
-
-func zipkinV1BinAnnotationsToOCAttributes(binAnnotations []*binaryAnnotation, parseStringTags bool) (attributes *tracepb.Span_Attributes, status *tracepb.Status, fallbackServiceName string) {
+func zipkinV1BinAnnotationsToOCAttributes(span ptrace.Span, binAnnotations []*binaryAnnotation, parseStringTags bool) string {
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/json.go]: Fixed, hopefully

File: pkg/translator/zipkin/zipkinv1/json.go (hunk 3 above)
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/json.go:216]: Done.

File: pkg/translator/zipkin/zipkinv1/json.go [hunk 6]
```diff
@@ -374,165 +293,126 @@ func parseZipkinV1Annotations(annotations []*annotation) *annotationParseResult
...
-
-	if attributeMap := endpoint.createAttributeMap(); attributeMap != nil {
-		req.Node.Attributes = attributeMap
-	}
-
-	m[nodeKey] = req
-
-	return req
+	rs := td.ResourceSpans().AppendEmpty()
+	rs.Resource().Attributes().PutString(conventions.AttributeServiceName, endpoint.ServiceName)
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply pkg/translator/zipkin/zipkinv1/json.go:374]: DOne.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**